src/components/VersionRestartModal/VersionRestartModal.tsx

```diff
@@ -62,37 +62,35 @@ const VersionRestartModal: React.FC<Props> = ({
     },
     refetchQueries,
   });
-  const { data } = useQuery<BuildVariantsQuery, BuildVariantsQueryVariables>(
-    GET_BUILD_VARIANTS,
-    {
-      variables: { id: versionId },
-    }
-  );
+
+  const { data } = useQuery<
+    BuildVariantsWithChildrenQuery,
+    BuildVariantsWithChildrenQueryVariables
+  >(GET_BUILD_VARIANTS_WITH_CHILDREN, {
+    variables: { id: versionId },
+  });
+
   const { version } = data || {};
-  const { buildVariants } = version || {};
+  const { buildVariants, childVersions } = version || {};
   const [
     selectedTasks,
     patchStatusFilterTerm,
     baseStatusFilterTerm,
     { toggleSelectedTask, setPatchStatusFilterTerm, setBaseStatusFilterTerm },
-  ] = usePatchStatusSelect(buildVariants);
-  const { search } = useLocation();
-  const prevSearch = usePrevious(search);
-  useEffect(() => {
-    if (search !== prevSearch) {
-      const urlParams = parseQueryString(search);
-      setPatchStatusFilterTerm(
-        getArray(urlParams[PatchTasksQueryParams.Statuses]) ?? []
-      );
-      setBaseStatusFilterTerm(
-        getArray(urlParams[PatchTasksQueryParams.BaseStatuses]) ?? []
-      );
-    }
-  }, [search, prevSearch, setBaseStatusFilterTerm, setPatchStatusFilterTerm]);
```

Comment:
Some users rely on this useEffect to allow preselecting restart modal checkboxes by interacting with the status filters from the tasks table. I think we should keep this feature to preselect restart modal checkboxes for parent patch tasks or at least discuss with design.

Reply:
I was wondering how these would work now that we have child patches. Where the tasks the user filters might not/will not correspond to the tasks present in the child patches.